packages/driver/src/cy/net-stubbing/events/response.ts

```diff
@@ -75,7 +75,14 @@ export const onResponse: HandlerFn<CyHttpMessages.IncomingResponse> = async (Cyp
         // arguments to res.send() are merged with the existing response
         const _staticResponse = _.defaults({}, staticResponse, _.pick(res, STATIC_RESPONSE_KEYS))
 
-        _.defaults(_staticResponse.headers, res.headers)
+        _staticResponse.headers = _.defaults({}, _staticResponse.headers, res.headers)
+
+        // https://github.com/cypress-io/cypress/issues/17084
+        // When a user didn't provide content-type,
+        // we remove the content-type provided by the server
+        if (!staticResponse.headers || !staticResponse.headers['content-type']) {
+          delete _staticResponse.headers['content-type']
+        }
```

Comment:
This code block is causing https://app.circleci.com/pipelines/github/cypress-io/cypress/22275/workflows/eb6fc6d5-607e-4bf1-8cc6-c17ae12b0e03/jobs/817100 to fail when attempting to override the content-type of a .csv. Check out this spec: https://github.com/cypress-io/cypress-example-recipes/blob/master/examples/testing-dom__download/cypress/integration/location-href-spec.js

Reply:
```suggestion
        _staticResponse.headers = _.defaults({}, _staticResponse.headers, res.headers)

        // https://github.com/cypress-io/cypress/issues/17084
        // When a user didn't provide content-type,
        // and they provided body as an object,
        // we remove the content-type provided by the server
        if (!staticResponse.headers || !staticResponse.headers['content-type']) {
          if (typeof _staticResponse.body === 'object') {
            delete _staticResponse.headers['content-type']
          }
        }
```

I don't have an authority to push commits to this branch. So, I leave this suggestion. 

These lines of code had to be added because when a user doesn't provide content-type, the server-provided content-type was used and the returned type of content was string when user provided the `object` body. 

But this removes legitimate `content-type`s provided by the server. So, I added a condition to remove the `content-type` only when user-provided `body` type is `object`.